test/buildAndRun.test.ts

```diff
@@ -0,0 +1,120 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+// The module 'assert' provides assertion methods from node
+import * as assert from 'assert';
+import * as assertEx from './assertEx';
+import * as vscode from 'vscode';
+import { commands } from 'vscode';
+import { Uri } from 'vscode';
+import * as fse from 'fs-extra';
+import * as AdmZip from 'adm-zip';
+import * as path from 'path';
+import { Platform } from "../configureWorkspace/config-utils";
+import { ext } from '../extensionVariables';
+import { Suite } from 'mocha';
+import { configure } from '../configureWorkspace/configure';
+import { TestUserInput } from 'vscode-azureextensionui';
+import { getTestRootFolder, constants } from './global.test';
+import { httpsRequestBinary } from '../utils/httpRequest';
+import { TestTerminalProvider } from '../commands/utils/TerminalProvider';
+
+let testRootFolder: string = getTestRootFolder();
+
+async function unzipFileFromUrl(uri: Uri, sourceFolderInZip: string, outputFolder: string): Promise<void> {
+    let zipContents = await httpsRequestBinary(uri.toString());
+    let zip = new AdmZip(zipContents);
+    await extractFolderTo(zip, sourceFolderInZip, outputFolder);
+}
+
+async function extractFolderTo(zip: AdmZip, sourceFolderInZip: string, outputFolder: string): Promise<void> {
```

Comment:
/**
 * Extracts only a specific folder and its subfolders.
 * Not using AdmZip.extractAllTo because depending on the .zip file we may end up with an extraneous top-level folder
 */